src/services/kvk/service.ts

```diff
@@ -76,7 +45,70 @@ export async function searchBedrijvenInHandelsRegister(
     searchParams.set("pagina", page.toString());
   }
 
-  return await searchInHandelsRegister(searchParams);
+  const url = `${zoekenUrl}?${searchParams}`;
+
+  return fetchLoggedIn(url).then(async (r) => {
+    if (r.status === 404) {
+      const body = await r.json();
+      if (hasFoutCode(body, "IPD5200")) return defaultPagination([]);
+    }
+    if (r.status === 400) {
+      throw new FriendlyError("Invalide zoekopdracht");
+    }
+    throwIfNotOk(r);
+    const body = await r.json();
+    return parseKvkPagination(body);
+  });
+}
+
+export function searchBedrijvenInHandelsRegisterByVestigingEnKvkNummer(
+  vestigingsnummer: string,
+  kvkNummer: string,
+  page?: number,
+) {
+  const searchParams = new URLSearchParams();
+  searchParams.set("kvkNummer", kvkNummer);
+  searchParams.set("vestigingsnummer", vestigingsnummer);
+  searchParams.set("type", "hoofdvestiging");
+  searchParams.append("type", "nevenvestiging");
+
+  return searchBedrijfInHandelsRegister(page, searchParams);
+}
+
+export function searchBedrijvenInHandelsRegisterByRsin(
+  rsin: string,
+  kvkNummer?: string,
+  page?: number,
+) {
+  const searchParams = new URLSearchParams();
+
+  if (rsin.length === 9) {
+    //ok2
+    searchParams.set("rsin", rsin);
+    if (kvkNummer) {
+      searchParams.set("kvkNummer", kvkNummer);
+    }
+  } else {
+    //e-suite variant van ok1
+    searchParams.set("kvkNummer", rsin);
```

Comment:
kunnen we er binnen deze refactoring voor zorgen dat we niet meer een kvknummer in een rsin property zetten? Of denk je dat dat teveel impact heeft?

Reply:
esuite gebruitk rsin nou eenmaal om het kvk nummer in op te slaan.

Reply:
Dat is niet helemaal waar. De esuite noemt het gewoon het kvk-nummer. In de openklant1-compatible api die ze hebben, heet het innNnpId en daar moet inderdaad het kvk-nummer in. Maar nergens wordt het rsin genoemd. Als we een rsin in een kvk veld stoppen dan doen we dat echt zelf. Als je zeker weet dat het op deze regel niet alleen maar nodig is vanwege de regel code bij https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/pull/1090#discussion_r2086395252, vind ik het ok om het in een andere story te doen.

Reply:
alleen al de discussie en verwarring maken het de moeite waard dit toch maar nu aan te pakken.  de aanpassingen staan klaar.